### DISCUSSION TITLE
Is it possible to use aggregations in Elastica adapter?

### DISCUSSION THREAD

**Discussion by codexp** [BabDev/Pagerfanta#52] [General]
I need to sum up a field on the ElasticaAdapter.
I am curious if it is possible to use aggregations with `ElasticaAdapter`?
Or are there any plans to implement that in future?

**Comment by mbabker** [comment]
Directly on the `ElasticaAdapter`, I'd have to see the actual use case here to decide if it fits in with the library.

If you need the aggregations to be part of your paginated dataset, you might be able to handle it already with one of the other adapters that have hook points (i.e. the callback or transforming adapters) or implement your own decorator around the `ElasticaAdapter`.  But, aggregations are not a feature I've personally worked with, so I'm not sure how those would fit in here (if at all).

**Comment by codexp** [comment]
After some inspection of the inner live, I`ve discovered one easy option:
Add aggregations in your query:
```php
$quantityAgg = new Aggregation\Sum('sum_quantity');
$quantityAgg->setField('quantity');

$query = (new Query())
    ->setQuery($yourQuery)
    ->addAggregation($quantityAgg)
;
```
Then on `ResultSet`:
```php
return [
    'agg' => [
        'sum_quantity' => $resultSet->getAggregation('sum_quantity')['value'] ?? 0,
    ],
];
```
